### PR TITLE
[BUGFIX] fix error getting validation result from DataContext

### LINKED ISSUE #3237: TypeError when getting latest validation result from DataContext
**Describe the bug**
I am trying to get the latest validation result from a `DataContext` using the `get_validation_result` method but am getting a

```
Traceback (most recent call last):
  File "C:\workspace\ge_tutorials\get_validation_result_error.py", line 10, in <module>
    latest_validation_result = data_context.get_validation_result("demo")
  File "C:\workspace\ge_tutorials\venv\lib\site-packages\great_expectations\data_context\data_context.py", line 2354, in get_validation_result
    filtered_key_list = sorted(filtered_key_list, key=lambda x: x.run_id)
TypeError: '<' not supported between instances of 'RunIdentifier' and 'RunIdentifier'
```

**To Reproduce**
1. [Clone `ge_tutorials`](https://docs.greatexpectations.io/docs/tutorials/getting_started/initialize_a_data_context#set-up-your-machine-for-the-tutorial) as per documentation tutorial
2. Run the following code at project root:
```python
import great_expectations as ge

expectations_root = "./getting_started_tutorial_final_v3_api/great_expectations"

data_context = ge.DataContext(context_root_dir=expectations_root)

# need multiple runs to compare
for _ in range(2):
    data_context.run_checkpoint(
        checkpoint_name="my_checkpoint"
    )
latest_validation_result = data_context.get_validation_result("demo")
print(latest_validation_result)
```

**Expected behavior**
When only the `expectation_suite_name` argument is provided the the `get_validation_result` should proceed to compare all of the validation results and return the latest, e.g. by comparing the timestamp-named subfolders alphabetically.

**Environment (please complete the following information):**
 - Operating System: Windows 10
 - Great Expectations Version: 0.13.26

**Additional context**
The problem boils down to a missing comparison dunder method implementation for the `RunIdentifier` class (or its super class). Currently only the `__eq__` / `__neq__` methods are implemented, at least one more, e.g. `__gt__` or `__lt__`, is need for the `sorted` operation [here](https://github.com/great-expectations/great_expectations/blob/d51afdd1af1ec1627cb5ee541847759255ce3bf1/great_expectations/data_context/data_context.py#L2274).

### PR DESCRIPTION
**Pull request by zachzIAM**
### Changes proposed in this pull request:
- add `total_ordering` to `DataContextKey` abstract class

### Previous Design Review notes:
- Closes #3237

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by zachzIAM** [comment]
@cla-bot check

File: tests/data_context/test_data_context.py [hunk 1]
```diff
@@ -390,6 +390,25 @@ def test_data_context_get_validation_result(titanic_data_context):
...
 
 
+def test_data_context_get_latest_validation_result(titanic_data_context):
+    """
+    Test that the latest validation result can be correctly fetched from the configured results
+    store
+    """
+    for _ in range(2):
+        titanic_data_context.profile_datasource("mydatasource")
+    assert len(titanic_data_context.validations_store.list_keys()) > 1
```
Comment (talagluck) [inline tests/data_context/test_data_context.py]: Could we change this to 
```suggestion
    assert len(titanic_data_context.validations_store.list_keys()) == 2
```
to be more explicit?

**Review by talagluck — COMMENTED** [review]
Thanks so much for this PR @zachzIAM! Overall, this implementation makes sense, and your test is understandable as well.

One request - in light of the [**Note** here](https://docs.python.org/3/library/functools.html#functools.total_ordering), I think it makes sense to just manually define a `__lt__` method (and add back the `__ne__`) instead of using `total_ordering`. Realistically, the performance difference will be negligible, but given that it is  straightforward to implement the `__lt__`, I think it will make the code a little bit more understandable.

I don't have a ton of experience with `total_ordering` so let me know if I'm off base here.

Thanks again!

**Comment by zachzIAM** [comment]
Hi @talagluck,

No issues implementing the individual dunders. I went for `total_ordering` to get the full set of comparators, implementing only `__lt__` (alongside `__eq__` and `__ne__`) means that the `>=` and `<=` operations won't work. Though we would still be ok for sorting so perhaps that's sufficient? 

Could also add a `__le__` implementation as well which should give us the `>=` and `<=` as well, think it's worth including?

**Comment by talagluck** [comment]
Thanks for the update, @zachzIAM! I agree that it is probably worth adding a `__le__` implementation as well just in case, even though it is unlikely that we will hit it.

Silly question, do we need `__gt__` and `__ge__` as well?

**Comment by zachzIAM** [comment]
> Silly question, do we need `__gt__` and `__ge__` as well?

@talagluck from documentation [here](https://docs.python.org/3/reference/datamodel.html#object.__ge__):

>There are no swapped-argument versions of these methods (to be used when the left argument does not support the operation but the right argument does); rather, `__lt__()` and `__gt__()` are each other’s reflection, `__le__()` and `__ge__()` are each other’s reflection, and `__eq__()` and `__ne__()` are their own reflection. If the operands are of different types, and right operand’s type is a direct or indirect subclass of the left operand’s type, the reflected method of the right operand has priority, otherwise the left operand’s method has priority...

So with the `__ge__` and `__gt__` implementations missing the comparison would still be done but via the `__le__` and `__lt__` implementations. Thinking about this, it seems better to be explicit and implement all 6 dunders like the Note you mentioned earlier suggests, and like the set up is currently with `__eq__` and `__ne__`.

**Review by talagluck — APPROVED** [review]
Thanks so much for being accommodating here, @zachzIAM! This looks great and ready to merge. Thanks again for your contribution!

**Pull request merged**